### ISSUE TITLE
 ./run_1_unpack_slc_topo_master  topo

### ISSUE THREAD

**Issue by alejobeap** [isce-framework/isce2#255]
Dear 
I change the fetchOrbits.py for the new, but at the moment that run the stack I have the error (bellow) in **Running: topo**, Before the code worked correctly, can you help me?

 ./run_1_unpack_slc_topo_master 
Completed Parsing the Configuration file
Functions to be executed:
['Function-1', 'Function-2']
Running: Sentinel1_TOPS
['--dirname', '/mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip', '--swaths', '2 1', '--orbit', '/mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/orbits/20200203/S1B_OPER_AUX_POEORB_OPOD_20200203T110527_V20200113T225942_20200115T005942.EOF', '--outdir', '/mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/master', '--auxdir', '/mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/Auxdir', '--bbox', '-0.45 -0.31 -91.61 -91.42', '--pol', 'vv']
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed sofware at
https://winsar.unavco.org/software/isce
Input XML files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/annotation/s1b-iw2-slc-vv-20200203t114921-20200203t114948-020104-0260db-005.xml']
Input TIFF files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/measurement/s1b-iw2-slc-vv-20200203t114921-20200203t114948-020104-0260db-005.tiff']
Manifest files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/manifest.safe']
MANS:  /mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/manifest.safe
Setting IPF version to :  003.20
Extracting orbit from Orbit File:  /mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/orbits/20200203/S1B_OPER_AUX_POEORB_OPOD_20200203T110527_V20200113T225942_20200115T005942.EOF
Time stamp (2020-02-03 11:49:22.607475) falls outside of the interpolation interval [9999-12-31 00:00:00:0001-01-01 00:00:00]
Input XML files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/annotation/s1b-iw1-slc-vv-20200203t114920-20200203t114947-020104-0260db-004.xml']
Input TIFF files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/measurement/s1b-iw1-slc-vv-20200203t114920-20200203t114947-020104-0260db-004.tiff']
Manifest files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/manifest.safe']
MANS:  /mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.zip S1B_IW_SLC__1SDV_20200203T114920_20200203T114949_020104_0260DB_102D.SAFE/manifest.safe
Setting IPF version to :  003.20
Extracting orbit from Orbit File:  /mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/orbits/20200203/S1B_OPER_AUX_POEORB_OPOD_20200203T110527_V20200113T225942_20200115T005942.EOF
Time stamp (2020-02-03 11:49:21.652668) falls outside of the interpolation interval [9999-12-31 00:00:00:0001-01-01 00:00:00]
**Running: topo**
['--master', '/mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/master', '--dem', '/mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/DEM/demLat_N00_S01_Lon_W092_W091.dem.wgs84', '--geom_masterDir', '/mnt/procesamiento/Procesamiento_Sentinel/Fernandina/Stack_Descendente/geom_master']
Traceback (most recent call last):
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 251, in <module>
    main(args.start,args.end)
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 242, in main
    cfgParser.runCmd()
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 54, in runCmd
    func_modules.main(self.funcParams[section])
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/topo.py", line 108, in main
    bbox = [np.min(boxes[:,0]), np.max(boxes[:,1]), np.min(boxes[:,2]), np.max(boxes[:,3])]
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

**Comment by bjmarfito** [comment]
I have the same issue after using the modified fetchOrbit.py. I've found out that fetchOrbit.py for precised orbits downloads the orbits files on the day they were processed not the the date were they could be applicable to the Sentinel-1 images. You could check my PR https://github.com/isce-framework/isce2/pull/257 where  I made some changes to the code.

Pls. give your feedback once you've used it.

Thanks!

**Comment by alejobeap** [comment]
Thanks I change the code, the orbits download is ok.
The problem continue in the ./run_1_unpack_slc_topo_master topo

I tried again but I had the same problem
Extracting orbit from Orbit File:  /mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/orbits/20210306/S1B_OPER_AUX_RESORB_OPOD_20210306T222839_V20210306T181614_20210306T213344.EOF
Time stamp (2021-03-06 10:52:22.231167) falls outside of the interpolation interval [9999-12-31 00:00:00:0001-01-01 00:00:00]


I see the orbit folder and the orbit is in the date (20200306 example)

Thanks

Alejo

**Comment by bjmarfito** [comment]
@alejobeap, thank you for your feedback. I've edited the fetchOrbit.py based on your comments. Pls, check if it works correctly now. https://github.com/isce-framework/isce2/pull/257/commits/c8af76be104bd51f8a8e613e80e17d77c4b4ac30

**Comment by alejobeap** [comment]
Dear 

I had this message in the two of my orbits that I downloaded:

Completed Parsing the Configuration file
Functions to be executed:
['Function-1']
Running: Sentinel1_TOPS
['--dirname', '/mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip /mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.zip', '--swaths', '3 2', '--orbit', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/orbits/20210306/S1B_OPER_AUX_RESORB_OPOD_20210306T222839_V20210306T181614_20210306T213344.EOF', '--outdir', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/slaves/20210306', '--auxdir', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/Auxdir', '--bbox', '-2.08 -1.93 -78.40 -78.28', '--pol', 'vv']
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed sofware at
https://winsar.unavco.org/software/isce
Input XML files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.SAFE/annotation/s1b-iw3-slc-vv-20210306t105218-20210306t105244-025893-03169a-006.xml', '/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.zip/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.SAFE/annotation/s1b-iw3-slc-vv-20210306t105243-20210306t105308-025893-03169a-006.xml']
Input TIFF files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.SAFE/measurement/s1b-iw3-slc-vv-20210306t105218-20210306t105244-025893-03169a-006.tiff', '/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.zip/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.SAFE/measurement/s1b-iw3-slc-vv-20210306t105243-20210306t105308-025893-03169a-006.tiff']
Manifest files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.SAFE/manifest.safe', '/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.zip/S1B_IW_SLC__1SDV_20210306T105243_20210306T105310_025893_03169A_7AB3.SAFE/manifest.safe']
Input XML files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.SAFE/annotation/s1b-iw3-slc-vv-20210306t105218-20210306t105244-025893-03169a-006.xml']
Input TIFF files:  ['/vsizip//mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.SAFE/measurement/s1b-iw3-slc-vv-20210306t105218-20210306t105244-025893-03169a-006.tiff']
MANS:  /mnt/datos_sentinel/Datos_Sentinel1/Sangay/Descendente/SLC/S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.SAFE/manifest.safe
Setting IPF version to :  003.31
**not well-formed (invalid token): line 59, column 18**


In the other orbits (that I had as backup) I did not get the same message. Only in the two orbits downloaded from February 28 and March 06. I also deleted the orbits and downloaded them all again and I got the same message from the orbit after February 15 or the image of the 15th. 

**not well-formed (invalid token): line 59, column 18**


I skipped this error and continued but in the run_4 step I got the same message:

**./run_4_overlap_geo2rdr_resample**

Running: geo2rdr
['--slave', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/slaves/20210306', '--master', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/master', '--geom_masterDir', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/geom_master', '--coregSLCdir', '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/coreg_slaves/20210306', '--overlap']
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed sofware at
https://winsar.unavco.org/software/isce
Editado por: Wilson Acero, Pedro A. Espin B. & Patricia Mothes
Instituto Geofisico-Escuela Politecnica Nacional (2020)
www.igepn.edu.ec
CPU mode
Traceback (most recent call last):
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 251, in <module>
    main(args.start,args.end)
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 242, in main
    cfgParser.runCmd()
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 54, in runCmd
    func_modules.main(self.funcParams[section])
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/geo2rdr.py", line 226, in main
    slave = ut.loadProduct(os.path.join(inps.slave, 'IW{0}.xml'.format(swath)))
  File "/home/insar/instaladores/isce2/contrib/stack/topsStack/s1a_isce_utils.py", line 26, in loadProduct
    obj = pm.loadProduct(xmlname)
  File "/home/insar/programas/isce_2019_09/isce/components/iscesys/Component/ProductManager.py", line 25, in loadProduct
    self.load(filename)
  File "/home/insar/programas/isce_2019_09/isce/components/iscesys/Component/Configurable.py", line 1410, in load
    tmpProp, tmpFact, tmpMisc = FP.parse(filename)
  File "/home/insar/programas/isce_2019_09/isce/components/iscesys/Parsers/XmlParser.py", line 41, in parse
    root = ET.parse(filename)
  File "/home/insar/anaconda3/envs/isce2/lib/python3.7/xml/etree/ElementTree.py", line 1197, in parse
    tree.parse(source, parser)
  File "/home/insar/anaconda3/envs/isce2/lib/python3.7/xml/etree/ElementTree.py", line 587, in parse
    source = open(source, "rb")
FileNotFoundError: [Errno 2] No such file or directory: '/mnt/procesamiento/Procesamiento_Sentinel/Sangay/Stack_Descendente/slaves/20210306/IW2.xml'

I think there is something wrong with the orbits since the end of February but I am not sure because without considering these dates the stack is working correctly.


I have run it in another place and I get the same error. 
MANS:  /mnt/datos_sentinel/Datos_Sentinel1/Fernandina/Descendente/SLC/S1B_IW_SLC__1SDV_20210305T114925_20210305T114955_025879_03162B_ECA8.zip S1B_IW_SLC__1SDV_20210305T114925_20210305T114955_025879_03162B_ECA8.SAFE/manifest.safe
Setting IPF version to :  003.31
not well-formed (invalid token): line 59, column 18

Thanks for your help.

**Comment by bjmarfito** [comment]
@alejobeap, thank you for your feedback. I'll test my edits on the fetchOrbit.py again in the coming days if there are errors that I've made but based on my prior testing the code correctly downloaded the orbit file associated with the image file. I'll update you regarding this.

The previous edits made on the fetchOrbit.py (PR https://github.com/isce-framework/isce2/pull/254) fails to account the appropriate time on the restituted orbits and the appropriate day on the precise orbits with respect to the image file. Hence, the downloaded orbits files don't match with the image files.

Based on your processing errors, I think you've downloaded the incorrect orbit file for your images and the issue is not on the information on the orbit file. One of your images named S1B_IW_SLC__1SDV_20210306T105218_20210306T105245_025893_03169A_0843.zip uses the orbit file with name S1B_OPER_AUX_RESORB_OPOD_20210306T222839_V20210306T181614_20210306T213344.EOF. The image filename shows the acquisition date and time is March 6, 2021 10:52:18 to 10:52:45 while your orbit filename shows that it is only applicable from March 6, 2021 18:16:14 to 21:33:44. Hence, the parameters on the orbit file is not applicable on the image file which causes the stack processing to fail. Additionally, Sentinel-1 InSAR images from the ongoing seismic activity in Iceland shows that the orbit information from ESA is trustworthy.

**Comment by alejobeap** [comment]
@bjmarfito Thanks for the information ad your help. I'm waiting for news

**Comment by bjmarfito** [comment]
@alejobeap, I received a confirmation from my colleague that the edited fetchOrbit.py ran smoothly on his computer. I think that you may have not properly incorporated the edited code on my previous pull request. You may download the edited code on this site (https://onedrive.live.com/?authkey=%21APZ5Z8oqV4jvyY8&cid=B5903921196601EA&id=B5903921196601EA%2163958&parId=B5903921196601EA%2152894&o=OneUp) until ISCE developers merge/review my pull request.

After downloading the edited code, replace the existing fetchOrbit.py on the topsStack folder.

Please give some feedback after you've used it.

Thanks!

**Comment by bjmarfito** [comment]
@alejobeap, just edited the code based on the comment of my colleague. You may now used the link on the previous comment.

**Comment by alejobeap** [comment]
@bjmarfito Thanks, I've already replaced it in my folder and I'm going to try it out. I will let you know about it.

**Comment by alejobeap** [comment]
@bjmarfito Thanks for your help. Now the stack is ok!! I have tested Sangay Volcano and it worked correctly with the latest image of March 06th. I will try with the 12th.

**Comment by bjmarfito** [comment]
Thank you also for your feedback @alejobeap.

**Comment by rbiessel** [comment]
This also fixed the problems I was having. Thank you!

**Comment by guangbaoshen** [comment]
i have met this problem also, hoping i will fixed it

**Comment by guangbaoshen** [comment]
Dear @bjmarfito
    I have met the same error "IndexError : too many indics for array: array is 1 dimensional, but 2 were indexed" when I tried to run the 'run_01_unpack_topo_reference' scripts.
    And I've checked the Orbits files  which I downloaded using fetchOrbit.py scripts. The time of the orbits file match with the image file.
    Is that anything wrong that I do not noticed? Appricated for your help!
    By the way, I am new here and I downloaded ISCE just few days ago, thank you very much!

**Comment by alejobeap** [comment]
Dear I update the fetchorbits.py (in issue #253 ). Now I'm running a stack, I'll let you know how it goes.

**Comment by bjmarfito** [comment]
@guangbaoshen, if you could post the whole error message like what @alejobeap here it would greatly help if the error you encounters is same as with him. Anyway, you can try this fetchOrbit.py (https://1drv.ms/u/s!AuoBZhkhOZC1g_NWps1cFSp3vljdFA?e=vUsFFt) and replace the one from the topsStack. I've made some huge changes since I've submitted my PR based on the feedback by @alejobeap and @rtburns-jpl. Pls. give me some feedback once you've used it.

Also pls. take note, as mentioned by @piyushrpt and @EJFielding, the latest orbit files are not taken account on this code because it uses the https://aux.sentinel1.eo.esa.int which has a typical delay in releasing orbit files and may be removed in the near future. Instead, if you want to process the latest Sentinel-1 image files, pls. use https://scihub.copernicus.eu/gnss/#/home to get the orbit files.

**Comment by guangbaoshen** [comment]
@bjmarfito Thank you very much for your reply.
Here I list the files in the 4 dictinoaries(AuxDir, SLC, DEM，Orbits) in case of giving the wrong files or dictionary when I run the scripts. Then I give the error message I got when I run the "SentinelWrapper.py". 
 
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE$ ^C
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE$ cd AuxDir/
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/AuxDir$ ls
data  manifest.safe  support
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/AuxDir$ cd SLC
bash: cd: SLC: No such file or directory
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/AuxDir$ cd ..
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE$ cd SLC
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/SLC$ ls
S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip
S1B_IW_SLC__1SDV_20180521T221211_20180521T221238_011025_014336_38A0.zip
S1B_IW_SLC__1SDV_20180602T221212_20180602T221239_011200_0148E0_4F05.zip
S1B_IW_SLC__1SDV_20180614T221213_20180614T221240_011375_014E44_06CB.zip
S1B_IW_SLC__1SDV_20180626T221213_20180626T221240_011550_0153AF_5390.zip
S1B_IW_SLC__1SDV_20180708T221214_20180708T221241_011725_01591E_0B8A.zip
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/SLC$ cd ..
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE$ cd DEM/
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/DEM$ ls
demLat_N38_N40_Lon_E116_E118.dem.wgs84      demLat_N38_N40_Lon_E116_E118.dem.wgs84.xml  isce.log
demLat_N38_N40_Lon_E116_E118.dem.wgs84.vrt  insar.log
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/DEM$ cd ..
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE$ cd Orbits/
(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE/Orbits$ ls
S1B_OPER_AUX_POEORB_OPOD_20180529T110601_V20180508T225942_20180510T005942.EOF
S1B_OPER_AUX_POEORB_OPOD_20180610T110539_V20180520T225942_20180522T005942.EOF
S1B_OPER_AUX_POEORB_OPOD_20180622T110648_V20180601T225942_20180603T005942.EOF
S1B_OPER_AUX_POEORB_OPOD_20180704T110539_V20180613T225942_20180615T005942.EOF
S1B_OPER_AUX_POEORB_OPOD_20180716T110549_V20180625T225942_20180627T005942.EOF
S1B_OPER_AUX_POEORB_OPOD_20180728T110605_V20180707T225942_20180709T005942.EOF



(isce2) guangbao@guangbao-BAD-INDEX:~/ISCE$ SentinelWrapper.py -c configs/config_reference 
/home/guangbao/anaconda3/envs/isce2/isce2/contrib/stack/topsStack/SentinelWrapper.py:113: SyntaxWarning: "is" with a literal. Did you mean "=="?
  if iString is '':
Completed Parsing the Configuration file
Functions to be executed:
['Function-1', 'Function-2']
Running: Sentinel1_TOPS
['--dirname', '/home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip', '--swaths', '1 2 3', '--orbitdir', '/home/guangbao/ISCE/Orbits', '--outdir', '/home/guangbao/ISCE/reference', '--auxdir', '/home/guangbao/ISCE/AuxDir', '--bbox', '38 39 116 117', '--pol', 'vv']
This is the Open Source version of ISCE.
Some of the workflows depend on a separate licensed package.
To obtain the licensed package, please make a request for ISCE
through the website: https://download.jpl.nasa.gov/ops/request/index.cfm.
Alternatively, if you are a member, or can become a member of WinSAR
you may be able to obtain access to a version of the licensed sofware at
https://winsar.unavco.org/software/isce
Input XML files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/annotation/s1b-iw1-slc-vv-20180509t221211-20180509t221236-010850-013d8b-004.xml']
Input TIFF files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/measurement/s1b-iw1-slc-vv-20180509t221211-20180509t221236-010850-013d8b-004.tiff']
Manifest files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/manifest.safe']
MANS:  /home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/manifest.safe
Setting IPF version to :  002.90
'xml.etree.ElementTree.Element' object has no attribute 'getchildren'
Input XML files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/annotation/s1b-iw2-slc-vv-20180509t221212-20180509t221237-010850-013d8b-005.xml']
Input TIFF files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/measurement/s1b-iw2-slc-vv-20180509t221212-20180509t221237-010850-013d8b-005.tiff']
Manifest files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/manifest.safe']
MANS:  /home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/manifest.safe
Setting IPF version to :  002.90
'xml.etree.ElementTree.Element' object has no attribute 'getchildren'
Input XML files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/annotation/s1b-iw3-slc-vv-20180509t221210-20180509t221236-010850-013d8b-006.xml']
Input TIFF files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/measurement/s1b-iw3-slc-vv-20180509t221210-20180509t221236-010850-013d8b-006.tiff']
Manifest files:  ['/vsizip//home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/manifest.safe']
MANS:  /home/guangbao/ISCE/SLC/S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.zip S1B_IW_SLC__1SDV_20180509T221210_20180509T221237_010850_013D8B_B303.SAFE/manifest.safe
Setting IPF version to :  002.90
'xml.etree.ElementTree.Element' object has no attribute 'getchildren'
Running: topo
['--reference', '/home/guangbao/ISCE/reference', '--dem', '/home/guangbao/ISCE/DEM/demLat_N38_N40_Lon_E116_E118.dem.wgs84', '--geom_referenceDir', '/home/guangbao/ISCE/geom_reference', '--numProcess', '1']
running in parallel with 1 processes
Traceback (most recent call last):
  File "/home/guangbao/anaconda3/envs/isce2/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 251, in <module>
    main(args.start,args.end)
  File "/home/guangbao/anaconda3/envs/isce2/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 242, in main
    cfgParser.runCmd()
  File "/home/guangbao/anaconda3/envs/isce2/isce2/contrib/stack/topsStack/SentinelWrapper.py", line 54, in runCmd
    func_modules.main(self.funcParams[section])
  File "/home/guangbao/anaconda3/envs/isce2/isce2/contrib/stack/topsStack/topo.py", line 112, in main
    bbox = [np.min(boxes[:,0]), np.max(boxes[:,1]), np.min(boxes[:,2]), np.max(boxes[:,3])]
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

**Comment by bjmarfito** [comment]
Thank you @guangbaoshen for your reply. I think your problem is different form this PR since it is concern with this PR is configuration files of stackSentinel.py running mismatched orbit files. Based from the orbit filename you have, I think the image and the orbit files match.

I think I've encountered the error 'xml.etree.ElementTree.Element' object has no attribute 'getchildren' before. This is somehow related to Python 3.9. Are you using this version of Python?

Anyway, I suggest you create a separate PR for this this since your concern is different from @alejobeap.

**Comment by bjmarfito** [comment]
@guangbaoshen I mean you need to create a separate issue for this. Apologies for the confusion. Thanks!

**Comment by guangbaoshen** [comment]
@bjmarfito Thank you very much, and the python version I am using is 3.9.2, I think it is the problem. I will create a seperate issue for this and I will let you know if the problem is fixed. Thank you very much for your kind help.

**Comment by EJFielding** [comment]
@bjmarfito It would be helpful if you submitted your changed version of `fetchOrbit.py` as a Pull Request from your GitHub account. Then it can be more easily used by others and considered for incorporation into the main ISCE2 tree.
